File: csharp/IceDiscovery/replication/HelloI.cs
// Copyright (c) ZeroC, Inc.

using Demo;

internal class HelloI(string name) : HelloDisp_
{
    public override string getGreeting(Ice.Current current)
    {
        return _name + " says Hello World!";
    }

    public override void shutdown(Ice.Current current)
    {
        System.Console.Out.WriteLine("Shutting down...");
        current.adapter.getCommunicator().shutdown();
    }

    private string _name = name;
}

File: cpp/IceGrid/replication/HelloI.cpp
// Copyright (c) ZeroC, Inc.

#include "HelloI.h"
#include <Ice/Ice.h>
#include <iostream>

using namespace std;

HelloI::HelloI(string_view name) : _name(name) {}

string
HelloI::getGreeting(const Ice::Current&)
{
    return _name + " says Hello World!";
}

void
HelloI::shutdown(const Ice::Current& current)
{
    cout << _name << " shutting down..." << endl;
    current.adapter->getCommunicator()->shutdown();
}
